platform/security/pep/security-pep-interceptor/src/main/java/ddf/security/pep/interceptor/SecurityAssertionStore.java

```diff
@@ -116,12 +117,16 @@ public static SecurityAssertion getSecurityAssertion(Message message) {
   }
 
   /**
-   * Return the TokenStore associated with this message.
+   * Return the TokenStore associated with this message
    *
    * @param message
-   * @return TokenStore
+   * @return the TokenStore, or null if it could not be retrieved
    */
   public static TokenStore getTokenStore(Message message) {
-    return TokenStoreUtils.getTokenStore(message);
+    try {
+      return TokenStoreUtils.getTokenStore(message);
+    } catch (TokenStoreException e) {
+      return null;
+    }
```

Comment:
The only code change for this PR 🎉 

If this is null, we just return a new SecurityAssertionSaml instance. I'm not sure if that's what we actually want, as the code looks like we were not expecting the token store to be null. Would appreciate another set of eyes.

https://github.com/codice/ddf/blob/master/platform/security/pep/security-pep-interceptor/src/main/java/ddf/security/pep/interceptor/SecurityAssertionStore.java#L115

Reply:
It will work correctly the way it is and the PEP still fails closed as long as the token for the assertion is null, which it will be if you just create an empty SAML assertion. However, it might be more correct if we just return an exception if the token store is null or in the case where we might flow down to returning that empty SecurityAssertionSaml instance. I think all that empty instance does is trigger the error condition anyways, so we may as well just jump there. That way we can avoid someone changing the PEP in the future and this possibly failing open at some point.

Reply:
Ok thanks for confirming. I'm inclined to keep it as-is in that case since the PEP still denies access. I checked and the PEP has tests for null tokens and null assertions so that should raise some flags if the it starts failing open.